opennms-doc/releasenotes/src/asciidoc/upgrade/upgrade-20-21.adoc

```diff
@@ -0,0 +1,45 @@
+[[upgrade-19-20]]
+== Upgrade OpenNMS Horizon 19 to 20
+
+=== Foreign Sources and Requisitions
+
+The Foreign Sources and Requisitions have been moved to the database.
+They must be manually migrated. In order to make the migration as smooth as possible a `foreignsource` and `requisition` osgi karaf command has been introduced.
```

Comment:
Added instructions.
Nothing would happen, as the requisition is not yet persisted.

The migration of each requisition is made in memory.

If an import is triggered before the migration is finished, the requisition does not yet exist and nothing should happen.

However if a requisition exists in active and pending mode, then an import may already been triggered, even if not yet both requisitions have been migrated. This scenario should probably be also considered.